From 76af6b45a5d8b8164ff2b8b9280a2dcebe1938cb Mon Sep 17 00:00:00 2001
From: Nishanth Menon <nm@ti.com>
Date: Mon, 28 Feb 2011 22:41:36 +0530
Subject: [PATCH 55/56] omap3+: smartreflex: enable/disable iff voltage data present

Dont blindly believe that volt_data entity exists. test before using

Signed-off-by: Nishanth Menon <nm@ti.com>
---
 arch/arm/mach-omap2/smartreflex.c |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/smartreflex.c b/arch/arm/mach-omap2/smartreflex.c
index 09f65a0..d417dcc 100644
--- a/arch/arm/mach-omap2/smartreflex.c
+++ b/arch/arm/mach-omap2/smartreflex.c
@@ -289,6 +289,7 @@ static void sr_set_regfields(struct omap_sr *sr)
 
 static void sr_start_vddautocomp(struct omap_sr *sr)
 {
+	struct omap_volt_data *volt_data;
 	if (!sr_class || !(sr_class->enable) || !(sr_class->configure)) {
 		dev_warn(&sr->pdev->dev,
 			"%s: smartreflex class driver not registered\n",
@@ -303,6 +304,14 @@ static void sr_start_vddautocomp(struct omap_sr *sr)
 		return;
 	}
 
+	volt_data = omap_voltage_get_nom_volt(sr->voltdm);
+	if (IS_ERR_OR_NULL(volt_data)) {
+		dev_err(&sr->pdev->dev,
+			"%s: SR[%d]volt domain not found!\n",
+			__func__, sr->srid);
+		return;
+	}
+
 	if (!sr_class->enable(sr->voltdm,
 			omap_voltage_get_nom_volt(sr->voltdm)))
 		sr->autocomp_active = true;
@@ -318,6 +327,15 @@ static void sr_stop_vddautocomp(struct omap_sr *sr)
 	}
 
 	if (sr->autocomp_active) {
+		struct omap_volt_data *volt_data;
+
+		volt_data = omap_voltage_get_nom_volt(sr->voltdm);
+		if (IS_ERR_OR_NULL(volt_data)) {
+			dev_err(&sr->pdev->dev,
+				"%s: SR[%d]volt domain not found!\n",
+				__func__, sr->srid);
+			return;
+		}
 		sr_class->disable(sr->voltdm,
 				omap_voltage_get_nom_volt(sr->voltdm),
 				1);
-- 
1.7.1

